tests/memealyzer.spec.ts

```diff
@@ -0,0 +1,62 @@
+import { test, expect, WebSocket } from "@playwright/test";
+
+var uri: string = "http://localhost:1080/";
+
+interface Image {
+  id: string;
+  Id: string;
+}
+
+interface ImageEvent {
+  arguments: Image[];
+}
+
+test("Add Meme Test", async ({ page }) => {
+  await page.goto(uri);
+
+  let webSocket: WebSocket;
+
+  page.on("websocket", (ws) => {
+    webSocket = ws;
+  });
+
+  const add = page.locator(".form-row button").first();
+
+  await page.waitForSelector(".spinner-grow", { state: "detached" });
+
+  const cards = page.locator(".card");
+
+  const beforeCardCount = await cards.count();
+
+  let expectedId: string;
+  let actualId: string;
+
+  await Promise.all([
+    page
+      .waitForResponse(
+        (response) =>
+          response.url().endsWith("/image") && response.status() === 200
+      )
+      .then((response) => response.text())
+      .then((text) => {
+        const image = JSON.parse(text) as Image;
+        expectedId = image.id;
+      }),
+    await add.click(),
+  ]);
```

Comment:
I think this is fine, but some would prefer if this was broken down into async/await. Something like this:

```ts
  const promise = page.waitForResponse((response) =>
    response.url().endsWith("/image") && response.status() === 200);

  // If I understand correctly, we don’t need to wait for this.
  add.click();
  
  const response = await promise;
  const text = response.text();
  const image = JSON.parse(text) as Image;
  expectedId = image.id;
```

Reply:
But, again, Promise.all is fine :)